langfilter/langfilter.php

```diff
@@ -119,11 +119,13 @@ function langfilter_prepare_body_content_filter(App $a, &$hook_data)
 		return;
 	}
 
-	if (!empty($hook_data['item']['rendered-html'])) {
-		$naked_body = strip_tags($hook_data['item']['rendered-html']);
-	} else {
-		$naked_body = BBCode::toPlaintext($hook_data['item']['body'], false);
-	}
+	$naked_body = strip_tags(
+		$hook_data['item']['rendered-html']
+		??''?: // Equivalent of !empty()
+		BBCode::convert($hook_data['item']['body'], false, BBCode::INTERNAL, true)
```

Comment:
The previously used function `toPlaintext` is removing pictures in advance. This speeds up generating the BBCode. By directly using the `BBCode::convert` function the rendering of a page will drop measurably.

Reply:
Nope, the `BBCode::toPlaintext` runs a `BBCode::convert` first, then `HTML::toPlaintext`, so we're actually saving cycles while not keeping link URLs like `BBCode::toPlaintext` rightfully does.

Reply:
The `BBCode::toPlaintext` removes images before doing the whole processing:
https://github.com/friendica/friendica/blob/2021.12-rc/src/Content/Text/BBCode.php#L452-L454

Reply:
Images, yes, link URLs, no. The new method is still cheaper.

Reply:
But for images we have to call the proxy functionality which **really** has got a performance impact.

Reply:
It looks like this can be circumvented by choosing anything else than `INTERNAL` or `API` for `$simple_html`, I'll have a look at the other choices.

Reply:
See https://github.com/friendica/friendica-addons/pull/1230